app/views/admin/jobs.html

```diff
@@ -42,9 +42,27 @@
 		<p>No jobs have been run yet.</p>
 	#{/else}
 %{ } }%
+%{ def renderHeaderWithToc = { Map jobName, Map jobData ->
+	if (jobName.size() != jobData.size() || jobName.size == 0) {
+		return
+	}
+	def toc = '<span><ul>\n'
+	def tables = ''
+	jobName.each { id, name ->
+		toc += "<li><a href=\"#${id}\">${name}</a></li>\n"
+	}
+	toc += '\n</ul></span>'
+
+	out.print(toc)
+
+	jobData.each { id, data ->
+		out.print('<h2 id="' + id + '">' + jobName[id] + '<small>&nbsp;<a href="' + play.mvc.Router.reverse('Admin.runJobManually', [type: id]).url + '">Run</a></small></h2>\n')
+		renderTable(data)
+	}
+}
+%}
 <span class="pull-right">Time: ${new Date()}</span>
-<h2>Poller Jobs <small><a href="@{Admin.runJobManually('poller')}">Run</a></small></h2>
-${renderTable(pollerJobs)}
+%{renderHeaderWithToc(jobNames, jobDatas) %}
```

Comment:
That NPE is caused by this change.
The `pollerJobs` -> `jobNames, jobDatas` is just the half of the whole change.

The change from `Admin.java` is missing here. :confused: